sql/core/src/main/scala/org/apache/spark/sql/execution/SparkQl.scala

```diff
@@ -369,13 +369,33 @@ private[sql] class SparkQl(conf: ParserConf = SimpleParserConf()) extends Cataly
       case Token("TOK_CLEARCACHE", Nil) =>
         ClearCacheCommand
 
+      // SHOW TABLES [(IN|FROM) database_name] ['identifier_with_wildcards']
       case Token("TOK_SHOWTABLES", args) =>
-        val databaseName = args match {
-          case Nil => None
-          case Token("TOK_FROM", Token(dbName, Nil) :: Nil) :: Nil => Option(dbName)
+        // Parses the following conditions
+        // 1. SHOW TABLES
+        // 2. SHOW TABLES FROM database 'table_pattern'
+        // 3. SHOW TABLES FROM database
```

Comment:
why accept `FROM`? It seems kind of weird. Hive doesn't do this either: https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-ShowTables